docker/pravega/scripts/init_kubernetes.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+#
+# Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+
+k8() {
+    local namespace=$1
+    local resource_type=$2
+    local resource_name=$3
+    local jsonpath=$4
+    local bearer=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+    local cacert="/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+    if [ -z "$namespace" ]; then
+        retval=$( curl --cacert ${cacert} -H "Authorization: Bearer ${bearer}" https://kubernetes/api/v1/${resource_type}/${resource_name} 2> /dev/null | jq -rM "${jsonpath}" 2> /dev/null )
+    else
+        retval=$( curl --cacert ${cacert} -H "Authorization: Bearer ${bearer}" https://kubernetes/api/v1/namespaces/${namespace}/${resource_type}/${resource_name} 2> /dev/null | jq -rM "${jsonpath}" 2> /dev/null )
+    fi
+
+    echo "$retval"
+}
+
+init_kubernetes() {
+    if [ -z "${POD_NAMESPACE}" ]; then
+      echo "POD_NAMESPACE variable not set. Exiting..."
+      exit 1
+    fi
+
+    if [ -z "${POD_NAME}" ]; then
+      echo "POD_NAME variable not set. Exiting..."
+      exit 1
+    fi
+
+    local ns=${POD_NAMESPACE}
+    local podname=${POD_NAME}
+
+    if [ -d "/var/run/secrets/kubernetes.io" ] && [ ! -z "${K8_EXTERNAL_ACCESS}" ]; then
+        echo "Running in a Kubernetes environment and managed by the Pravega Operator with external access enabled"
+        echo "Trying to obtain the external service endpoint..."
+
+        service_type=$( k8 "${ns}" "services" "${podname}" ".spec.type" )
+        if [ "${service_type}" == "LoadBalancer" ]; then
+            export PUBLISHED_ADDRESS=$( k8 "${ns}" "services" "${podname}" ".status.loadBalancer.ingress[].ip" )
```

Comment:
Is there a possibility for multiple ips? Should we handle that scenario too?
ref : https://github.com/kubernetes-client/java/blob/e43a7d160e12ded0eeed033e5b43d32fb8c8d368/kubernetes/docs/V1LoadBalancerStatus.md

Reply:
Good observation. Load Balancer services definitely support for multiple external IP addresses, but AFAIK the main K8 platforms (GKE, PKS, EKS) do not provide more than one IP address. Moreover, we cannot configure more than one published IP address in Pravega.